articles/10-tips-for-working-with-node.js/index.md

```diff
@@ -0,0 +1,368 @@
+with just a single line command. To use Npm you need Node.js to be
+installed in your gadget. For OS X and Window OS, the Node.js installer
+has an easy to use interface and Npm is included with Node.js
+Installation. Developers using Linux need to run the following command
+inapt package managed to set up Node.js and Npm
+
+```
+sudo apt-get update
+
+sudo apt-get install node
+
+sudo apt-get install npm
+```
+
+You can then verify Npm installation by writing the following in your
+terminal `npm -v` or if you realize you have an older version run `npm
+install npm -g` (for windows) or `npm install -g npm\@latest` (for Linus
+and mac) to update to the current Npm version. Npm helps to install
+Node.js modules. For example, you just need to write `npm install`
+followed by your specified module ie 
+`npm install express`, once the
+package is installed you will be able to use `require ()` on your Node.js
+project and they will run just as if they were built-ins. Another
+important aspect of Npm is dependency management. You can run `npm
+install` in your root folder and any listed dependency in your
+package.json (if you have one) will be installed. This help mostly when
+installing Node.js project from git repository For example `vow` (Node.js
+testing framework) can be installed from git and its single dependency
+`eye` can automatically be installed ie
+
+```
+git clone https://github.com/clonehad/vows.git
+
+cd vows
+
+npm install
+```
+
+### 2.  Understand A to Z Most Useful Node.js. Modules and Frameworks
+
+The power of Node.js lies in hundreds of packages that make it easier to
+write your program. The packages are open source and they are the
+ecosystem of Node.js applications. Every Node.js developer will always
+interact with these packages to get a more efficient development
+workflow.
+
+We collect a list of some of the most useful libraries that will help
+you ease the development of Node.js applications.
+
+-   [Express](http://expressjs.com/): it a web application framework for
+    Node.js that handle robust APIs and Web servers
+
+-   [Gulp](https://www.npmjs.com/package/gulp): helps to automate
+    development workflow. It helps to automate simple tasks in the
+    development of web applications.
+
+-   [Moment](https://www.npmjs.com/package/moment): An insubstantial
+    JavaScript date library for parsing, validating, manipulating, and
+    formatting dates.
+
+-   [Async](https://caolan.github.io/async/v3/): provides powerful
+    straightforward Over 70 functions for working with asynchronous
+    JavaScript.
+
+-   [Lodash](https://lodash.com/): makes JavaScript easier by taking out
+    the irritation of working with arrays, numbers, objects, strings,
+    etc.
+
+-   [Jshint](https://www.npmjs.com/package/jshint): Helps to detect
+    errors and potential problems and threats in your Node.js
+    (JavaScript) code in the way you want your code to execute
```

Comment:
Consider adjusting to errors, potential problems in your Node.js (JavaScript) code in the way your code executes.